pkg/clientcert/controller_test.go

```diff
@@ -183,10 +183,13 @@ func TestSync(t *testing.T) {
 			controller := &clientCertificateController{
 				ClientCertOption: clientCertOption,
 				CSROption:        csrOption,
-				hubCSRLister:     hubInformerFactory.Certificates().V1().CertificateSigningRequests().Lister(),
-				hubCSRClient:     hubKubeClient.CertificatesV1().CertificateSigningRequests(),
-				spokeCoreClient:  agentKubeClient.CoreV1(),
-				controllerName:   "test-agent",
+				csrControl: &v1CSRControl{
```

Comment:
I think you would need some additional test for v1beta1 csr.

Reply:
added more tests, PTAL